src/Symfony/Component/Messenger/Transport/Schedule/Locator/ChainScheduleLocator.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Schedule\Locator;
+
+use Psr\Container\NotFoundExceptionInterface;
+use Symfony\Component\Messenger\Transport\Schedule\ScheduleConfig;
+
+final class ChainScheduleLocator implements ScheduleLocatorInterface
+{
+    /**
+     * @var ScheduleLocatorInterface[]
+     */
+    private array $locators;
+
+    private array $lastFound = [];
+
+    /**
+     * @param iterable<ScheduleLocatorInterface> $locators
+     */
+    public function __construct(iterable $locators)
```

Comment:
```suggestion
    public function __construct(
        private iterable $locators,
    )
```

Reply:
It won't work like that. The "itarable" is iterated and `add()` method is called for types checking.